MusicNotationKit/MusicNotationKit/Note.swift

```diff
@@ -53,3 +53,34 @@ extension Note: NoteCollection {
 	
 	internal var noteCount: Int { return 1 }
 }
+
+extension Note: CustomDebugStringConvertible {
+	public var debugDescription: String {
+		
+		// Tones are handled in two different ways. Single notes
+		// do not contain collection notation, whereas chords do.
+		var tones:String = ""
+		if (self.tones.count == 1) {
+			tones = String(self.tones[0])
+		} else if (self.tones.count > 1) {
+			tones = String(self.tones)
+		}
+		
+		// The dot notation is placed after the note duration.
+		var duration  = String(self.noteDuration.rawValue)
+		if (self.dot != nil) {
+			duration += String(self.dot!.rawValue)
+		}
+		
+		var result = String(format: "%@%@",
+			duration,
+			tones)
+		
+		if (self.isRest) {
+			result += "R"
+		}
```

Comment:
Combine these using string interpolation and the ternary operator so you can just return this value without having the temporary value.